src/zimscraperlib/imaging.py

```diff
@@ -68,15 +68,32 @@ def resize_image(
         else:
             resized = resizeimage.resize(method, image, [width, height])
 
+    # remove alpha layer if not supported and added during resizing
+    alpha_not_supported = ["JPEG", "BMP", "EPS", "PCX"]
+    if resized.mode == "RGBA" and image_format in alpha_not_supported:
+        resized = resized.convert(image_mode)
     # save the image
-    if resized.mode == "RGBA" and image_format == "JPEG":
-        resized = resized.convert("RGB")
     kwargs = {"JPEG": {"quality": 100}, "PNG": {}}
     resized.save(
-        str(to) if to is not None else fpath, image_format, **kwargs.get(image_format)
+        str(to) if to is not None else fpath,
+        image_format,
+        **kwargs.get(image_format, {}),
     )
 
 
+def change_image_format(src, dst, target_format, colorspace=None):
+    """ convert an image file from one format to another 
+
+        colorspace: RGB, ARGB, CMYK (and other PIL colorspaces)
+        target_format: JPEG, PNG, BMP (and other PIL formats) """
+    image = PIL.Image.open(src)
+    alpha_not_supported = ["JPEG", "BMP", "EPS", "PCX"]
+    if (image.mode == "RGBA" and target_format in alpha_not_supported) or colorspace:
+        image = image.convert("RGB") if not colorspace else image.convert(colorspace)
+    kwargs = {"JPEG": {"quality": 100}, "PNG": {}}
```

Comment:
should we factorize this?

should we allow additional args so this can be overriden ? Users might want to limit output quality on JPEG without going full-optimization. We could also consider we just change format here in a lossless manner and we may include optimization tools in the future.

Reply:
Did it